driver/modern_bpf/helpers/base/push_data.h

```diff
@@ -253,11 +256,20 @@ static __always_inline u16 push__charbuf(u8 *data, u64 *payload_pos, unsigned lo
  */
 static __always_inline u16 push__bytebuf(u8 *data, u64 *payload_pos, unsigned long bytebuf_pointer, u16 len_to_read, enum read_memory mem)
 {
+	u16 total_len = 0;
+	if (len_to_read > ARGS_ENV_SIZE_MAX)
```

Comment:
only a nit: Maybe it is worth to have some `min`/`max` macros here? (If so, maybe a follow-up PR).

Reply:
@hbrueckner could skip reading at all if len_to_read = 0

Reply:
@hbrueckner pushed a simple suggestion, let me know, can adjust it.

Reply:
> @hbrueckner could skip reading at all if len_to_read = 0
The code is already doing it, if `len_to_read=0` we push `0`

Reply:
Added benefit in the modern_bpf API setup is that we exit faster here. Thought @hbrueckner raised a valid point, always be consistent and do min/max checks if you do max checks.

Reply:
Uhm let's say here there is the usual trade-off:
* save one `bpf_probe_read` in case the `len_to_read` is `0`. (not frequent scenario)
* add complexity and pay for the checks when `len_to_read` is `!=0`. (frequent scenario). 

That's exactly the trade-off we have in the old probe, very complex code to save corner cases like this, to be honest, I would avoid this check at least for now, if we notice a real benefit from this we can always introduce it in a second step :thinking: